src/App.js

```diff
@@ -24,6 +25,8 @@ function App() {
           <Route element={<AuthenticateUser />}>
             <Route path='/posts' element={<PostsPage />} />
           </Route>
+          <Route path='/edit-profile' element={<EditProfile />} />
```

Comment:
These routes should be nested inside the `<Route element={<AuthenticateUser />}>`